### PR TITLE
Merge with master and fix test_api.py

### PR DESCRIPTION
**Pull request by echarles**
this branch resolves:

- the current merge issue with master on jupyter_server/services/kernels/handlers.py 
- fix compilation issue on NATIVE_KERNEL_NAME on tests/services/kernels/test_api.py

On my local env, test_api.py is failing with `jupyter_kernel_mgmt.kernelspec.NoSuchKernel: No such kernel named python3`

### CODE REVIEW

File: tests/services/kernels/test_api.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from jupyter_server.utils import url_path_join
 from ...utils import expected_http_error
 
+from ipython_genutils.py3compat import PY3
+NATIVE_KERNEL_NAME = 'python3' if PY3 else 'python2'
```
Comment (kevin-bates) [inline tests/services/kernels/test_api.py]: This is removed as part of the JKM changes.  NATIVE_KERNEL_NAME no longer exists.

File: tests/services/kernels/test_api.py [hunk 2]
```diff
@@ -136,6 +138,18 @@ def client_fetch(*parts, headers={}, params={}, **kwargs):
     assert restarted_kernel['id'] == kernel2['id']
     assert restarted_kernel['name'] == kernel2['name']
 
+    # Start a kernel with a path
+    r = await fetch(
+        'api', 'kernels',
+        method='POST',
+                body=json.dumps({
+            'name': NATIVE_KERNEL_NAME,
```
Comment (kevin-bates) [inline tests/services/kernels/test_api.py]: Use `'pyimport/kernel'` for the NATIVE_KERNEL_NAME.

**Review by kevin-bates — COMMENTED** [review]
Thanks for doing this Eric.  NATIVE_KERNEL_NAME is now obsolete.

**Comment by kevin-bates** [comment]
Hmm - the appveyor issue is odd.  It's unable to update jupyter_core as best as I can tell. ??

File: tests/services/kernels/test_api.py (hunk 1 above)
Comment (echarles) [reply tests/services/kernels/test_api.py]: Thx. removed that bit.

File: tests/services/kernels/test_api.py (hunk 2 above)
Comment (echarles) [reply tests/services/kernels/test_api.py]: Thx. This change makes the test green (or at least yellow) on my latptop.

**Comment by echarles** [comment]
yep, only failing py35 on appveyor due to a permission while installing libs. I don't see any way to retrigger the appveyor build apart from pushing a fake commit.

**Comment by kevin-bates** [comment]
In comparing your branch with the rebase/master that I'm working on, I see that yours doesn't include this commit: https://github.com/jupyter/jupyter_server/commit/0d67d5d5c6df64a6383e0b0211b23b6a536d3cf0

**Comment by echarles** [comment]
true, I realize I didn't pull the very latest master. Now latest is merged. Let's see what appveyor says.

**Comment by echarles** [comment]
Strange... Only on PY35. Should we install with `---user`?

```
ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\pip-uninstall-7v3f2fm2\\jupyter-migrate.exe'
Consider using the `--user` option or check the permissions.
Command exited with code 1
```

**Comment by kevin-bates** [comment]
Have you confirmed that the other python versions pass?  All I see is that they were cancelled.  Perhaps removing 3.5 from the list (temporarily) would provide a good datapoint before introducing `--user`.  But if 3.5 is the only one failing, then I would agree - although I'm not python-savy enough to understand all the ramifications about that.

**Comment by echarles** [comment]
True, other than 3.5 are cancelled. Note the windows on github actions succeed. Would it be a temporary issue with appveyor? Let's have a look tomorrow and maybe also how the other PRs behave.

**Comment by kevin-bates** [comment]
Hi Eric - I ran a couple experiments using PR #2.  If I remove 3.5 from the appveyor matrix, the other pythons pass.  I suspect the underlying issue is due to the initial jupyter_client install (that occurs when the conda env is created) is installing jupyter_core 4.5.0.  But JKM wants jupyter_core 4.6.x, so when installing Jupyter Server, it goes to cleanup jupyter_core 4.5.0 and, for whatever reason, runs into the issue with jupyter-migrate (which is an application from the core package).

Seems like we should decide if we're bailing on appveyor (and travis?) in favor of GH actions.  Or we could simply remove 3.5 from the appveyor matrix, knowing that Windows 3.5 is tested via GH actions.

**Comment by echarles** [comment]
I usually do like putting all the eggs in the same basket, but in this case IMHO github actions is reliable enough to remove appveyor and travis. They have given more disavantages than advantages so far.

I had already opened  https://github.com/jupyter/jupyter_server/pull/148 to remove travis. We can continue the conversation there and see if there are any objection for that.

**Comment by echarles** [comment]
Merged with latest master and build is still green (you can forget about appveyor which has been removed).

**Comment by kevin-bates** [comment]
Thank you Eric!  I'm really sorry I've haven't had the time to give this proper attention.

**Comment by echarles** [comment]
@kevin-bates Closing as discussed. We will rebase on master later.

**Pull request closed without merging**